flexget/plugins/filter/torrent_match.py

```diff
@@ -0,0 +1,185 @@
+                    log.warning('Input %s did not return anything', input_name)
+                    continue
+
+                for entry in result:
+                    urls = ([entry['url']] if entry.get('url') else []) + entry.get('urls', [])
+                    if any(url in entry_urls for url in urls):
+                        log.debug('URL for `%s` already in entry list, skipping.', entry['title'])
+                        continue
+
+                    entries.add(entry)
+                    entry_urls.update(urls)
+        return entries
+
+    def get_local_files(self, config, task):
+        cwd = os.getcwd()
+        entries = self.execute_inputs(config, task)
+        for entry in entries:
+            location = entry.get('location')
+            if not location or not os.path.exists(location):
+                log.warning('%s is not a local file. Skipping.', entry['title'])
+                entry.reject('not a local file')
+
+            if os.path.isfile(location):
+                entry['files_root'] = ''
+                entry['files']([
+                    TorrentMatchFile(os.path.basename(location), os.path.getsize(location))
+                ])
+            elif os.path.isdir(location):
+                # change working dir to make things simpler
+                os.chdir(location)
+                entry['files_root'] = os.path.basename(location)
+                entry['files'] = []
+                # traverse the file tree
+                for root, _, files in os.walk('.'):
+                    stripped_root = root.lstrip('.')  # remove the dot
+                    for f in files:
+                        file_path = os.path.join(stripped_root, f)
+                        entry['files'].append(
+                            TorrentMatchFile(os.path.join(entry['files_root'], file_path), os.path.getsize(file_path))
+                        )
+            else:
+                log.error('Does this happen?')
+            os.chdir(cwd)
+
+        return entries
+
+    # run first to ensure we have temp files so that torrent plugin can do its magic
+    @plugin.priority(256)
+    def on_task_modify(self, task, config):
+        for entry in task.accepted:
+            if 'torrent' not in entry and 'download' not in task.config:
+                # If the download plugin is not enabled, we need to call it to get
+                # our temp .torrent files
+                download = plugin.get_plugin_by_name('download')
+                download.instance.get_temp_files(task, handle_magnets=True, fail_html=True)
+
+    def prepare_config(self, config):
+        if not isinstance(config['max_size_difference'], float):
+            config['max_size_difference'] = float(config['max_size_difference'].rstrip('%'))
+
+        return config
+
+    # Run first, this is not really an output plugin though, but we need 'torrent' field, which is set in modify
+    @plugin.priority(255)
+    def on_task_output(self, task, config):
```

Comment:
Should this run _last_ in modify phase instead?

Reply:
Yes